diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -1431,3 +1436,30 @@ void DifferentialFunctionComparator::findDifference(
         }
     }
 }
+
+bool DifferentialFunctionComparator::equal(Instruction *InstL,
+                                           Instruction *InstR) {
+    const auto *ValL = dyn_cast<Value>(InstL);
+    const auto *ValR = dyn_cast<Value>(InstR);
+
+    if (!ValL || !ValR)
+        return false;
+
+    DenseMapIterator<const Value *, int> L = sn_mapL.find(ValL);
+    DenseMapIterator<const Value *, int> R = sn_mapR.find(ValR);
+
+    if (L == sn_mapL.end() && R == sn_mapR.end())
+        // Values are not in maps
+        return false;
+
+    if (L->second == sn_mapL.size() - 1 && R->second == sn_mapR.size() - 1)
+        // The last inserted values with the same serial number
+        // are the differing ones
+        return false;
```

Comment:
I think that this can be replaced by checking if the pair `{InstL, InstR}` is equal to `DifferingInstructions`. This would be also much more readable.